### PR TITLE
Break apart tests for unsafe_validate_unique/3

### PR DESCRIPTION
**Pull request by nathanl**
Refactor in response to https://github.com/elixir-ecto/ecto/pull/2905#discussion_r253647018

### CODE REVIEW

File: test/support/mock_repo.exs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+defmodule Ecto.MockRepo do
```
Comment (nathanl) [inline test/support/mock_repo.exs]: I wasn't sure about adding another `test/support` module, but it's very generic and this seemed cleaner to me than defining it in the test. I'm happy to move it to the test if that seems better to others.

Comment (josevalim) [reply test/support/mock_repo.exs]: Yeah, I would prefer to keep it inside the describe in the test case, as it is not used anywhere else. Even it was, having local fixtures is often better for clarity. :)

**Comment by josevalim** [comment]
This is fantastic, thank  you @nathanl! One comment and we should be good to go! :heart:

**Comment by nathanl** [comment]
@josevalim Updated!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**